### PR TITLE
[AKS] `az aks create/update`: Udpate RP registration code to work on azure monitor subscription

### PR DESCRIPTION
**Pull request by bragi92**
**Related command**

```az aks create -n kaveeshcli22 -g kaveeshcli --location westeurope --enable-azure-monitor-metrics --azure-monitor-workspace-resource-id "{full_id}" --grafana-resource-id "{full_id}"```

```az aks create -n kaveeshcli22 -g kaveeshcli --disable-azure-monitor-metrics ```

```az aks update -n kaveeshcli22 -g kaveeshcli --enable-azure-monitor-metrics --enable-windows-recording-rules```

**Description**
I am updating the rp registration logic to register the RP on the azure monitor workspace subscription if its provided as a parameter to the command and if it is not provided then it should work on the clusters subscription.

**Testing Guide**

**Scenario (RP registration)**:
- Use the command with and without the azure-monitor-workspace-resource-id parameter similar to `az aks create -n kaveeshcli22 -g kaveeshcli --location westeurope --enable-azure-monitor-metrics --grafana-resource-id "{full_id}"` and it should result in a success. (Try it on a subscription where the Microsoft.Monitor RP is not registered when not passing in the parameter)

**Scenario (default workspace creation)**:
- Use the command with and without the azure-monitor-workspace-resource-id parameter similar to `az aks create -n kaveeshcli22 -g kaveeshcli --location westeurope --enable-azure-monitor-metrics  --grafana-resource-id "{full_id}"` and it should result in a success.

**History Notes**

[AKS] `az aks create/update`: Udpate RP registration code to work on azure monitor subscription
[AKS] `az aks create/update`: Update to add default region for workspace creation in air gapped cloud

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/azuremonitormetrics/helper.py [hunk 1]
```diff
@@ -13,6 +13,19 @@
 )
 
 
+def get_subscription_id_from_resource_id(resource_id):
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/azuremonitormetrics/helper.py:16]: maybe you could use parse_resource_id (import from msrestazure.tools) to get the sub id

Comment (bragi92) [reply src/azure-cli/azure/cli/command_modules/acs/azuremonitormetrics/helper.py:16]: Makes sense, Thanks @FumingZhang I have updated the PR to reflect this.

**Review by FumingZhang — COMMENTED** [review]
lgtm

Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=90274930&view=results) to validate the change, test passed!
- test_aks_create_with_azuremonitormetrics
- test_aks_update_with_azuremonitormetrics

**Review by FumingZhang — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Could you please add some test cases for this PR change?

**Comment by bragi92** [comment]
Thank you for the suggestion @zhoxing-ms . I have added a test to check if the proper subscription is being used in the PR now.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**